lib/Backends/BackendUtils.cpp

```diff
@@ -15,30 +15,42 @@
  */
 #include "glow/Backends/BackendUtils.h"
 #include "glow/IR/Instrs.h"
+#include "llvm/Support/raw_ostream.h"
 
 using namespace glow;
 using llvm::cast;
 using llvm::isa;
 
 void glow::runtime::RuntimeBundle::collectConstants(const IRFunction *F) {
+  collectConstants(F->getGraph()->getParent());
+}
 
+void glow::runtime::RuntimeBundle::collectConstants(const Module *M) {
   // At compile time condense constants to a single block of memory.
   // This allows the graph to go away after compile time.
   // If there are no constants return nullptr.
   if (constantWeightVarsMemSize_ == 0) {
     constants_ = nullptr;
     return;
   }
+
   constants_ =
       (uint8_t *)alignedAlloc(constantWeightVarsMemSize_, TensorAlignment);
-  for (auto &v : F->getGraph()->getParent()->getConstants()) {
-    assert(isa<WeightVar>(F->getWeightForNode(v)));
-    auto *w = cast<WeightVar>(F->getWeightForNode(v));
-    auto payload = v->getPayload().getUnsafePtr();
-    auto numBytes = w->getSizeInBytes();
-    auto addr = getValueOffset(v);
+
+  for (const auto &symbol : symbolTable_) {
+    llvm::StringRef name = symbol.first;
+    const RuntimeSymbolInfo &info = symbol.second;
+
+    Constant *c = M->getConstantByName(name);
+    if (!c) {
```

Comment:
what is the situation when module does not have that constant?

Reply:
It's a consequence of the RuntimeBundle overlap issue you commented on in #2274, the SymbolTable contains both Constants and Placeholders. We only want to copy the contents of the Constants though, obviously.